### DISCUSSION TITLE
How can I make my own python file modules read by pyscript in a nested way?

### DISCUSSION THREAD

**Discussion by LucasCelestinoSE** [pyscript/pyscript#1951] [Q&A] [answered]
I have code that requires several nested dependencies imported into the same code.
Pyscript has pyodide as a dependency, and if I'm not mistaken, to make pyscript "read" all the codes and their nested dependencies, it is necessary to make a virtual machine and zip the file.
This is something that made me give up using only pyodide and move on to other options.

For you, community member, how can I make pyscript read code that has several dependencies??

Example:
---------------------------
```python
# ./Main.py
import arrr
from pyscript import document
import hello
def catch_argv(event):
  hello.say_hello
catch_argv(event)
```
-------------------------
```python
# ./Hello.py

def say_hello():
    print("Hello!")

````

Error mensager>
![Captura de tela de 2024-01-25 04-14-02](https://github.com/pyscript/pyscript/assets/98926627/af47825e-c36c-4744-b934-e8b3a031a1d3)

**Comment by WebReflection** [comment, marked as answer]
I took the freedom to edit your issue as it wasn't clear and it was using multi-line code within a single ` tick.

To bring in modules you can use [[files]](https://pyscript.github.io/docs/2023.12.1/user-guide/configuration/#files) which purpose is to load your files into the virtual FS automatically provided by this project.

That allows you to import `hello` as long as you have `hello.py` or an `hello` folder with its `__init__.py` in it, so that this issue is actually non-existent and, hopefully, well documented.

**Comment by LucasCelestinoSE** [comment]
Thank you really for helping me.
I apologize for my carelessness regarding ticks, it was early in the morning and I was a little frustrated.

**Reply by WebReflection** [reply]
I hate when it happens ... I can't work before any coffee in the morning ... no need to apologize, I just edited for future curious lurker or anyone with the same question 👍
